nvflare/app_common/executors/learner_executor.py

```diff
@@ -79,6 +79,8 @@ def execute(self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abort
                 return self.submit_model(shareable, fl_ctx)
             elif task_name == self.validate_task:
                 return self.validate(shareable, fl_ctx, abort_signal)
+            elif task_name == ValidateType.MODEL_VALIDATE_FOR_SEARCH:
+                return self.validate_for_search(shareable, fl_ctx, abort_signal)
```

Comment:
If we think MODEL_VALIDATION_FOR_SEARCH is a common pattern.

To be consistent with previous code, I would suggest change this part to

```
elif task_name == self.validate_task:
    shareable.set_header(AppConstants.VALIDATE_TYPE, ValidateType.MODEL_VALIDATE)
    return self.validate(shareable, fl_ctx, abort_signal)
elif task_name == self.validate_for_search_task:
    shareable.set_header(AppConstants.VALIDATE_TYPE, ValidateType.MODEL_VALIDATE_FOR_SEARCH)
    return self.validate(shareable, fl_ctx, abort_signal)

```

And have ``self.validate_for_search_task = AppConstants.TASK_VALIDATE_FOR_SEARCH``
And then in constants.py define that ``AppConstants.TASK_VALIDATE_FOR_SEARCH = `model_validate_for_hp_search` ``



This is a BIG if, if this is not common enough, I suggest we inherit the learner_executor with something called "autofedrl_executor" in a file "autofedrl_executor.py" or something and overwrites the "execute" function.

Reply:
Good point. Refactoring to use an inherited class.